File: Server/src/main/java/bgu/spl171/net/impl/TFTP/Commands/WriteRequest.java
package bgu.spl171.net.impl.TFTP.Commands;

import bgu.spl171.net.impl.TFTP.Commands.Responses.Response;

public class WriteRequest implements Command {

    String fileName;
    public WriteRequest(String fileName) {
        this.fileName = fileName;
    }

    public String getFileName() {
        return fileName;
    }

    @Override
    public short getOpCode() {
        return OpCode.WriteRequest.getValue();
    }

    @Override
    public Response execute() {
        return null;
    }
}

File: java to c++/WriteRequest.cpp
#include "WriteRequest.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						using Response = bgu::spl171::net::impl::TFTP::Commands::Responses::Response;

						WriteRequest::WriteRequest(const std::wstring &fileName)
						{
							this->fileName = fileName;
						}

						std::wstring WriteRequest::getFileName()
						{
							return fileName;
						}

						short WriteRequest::getOpCode()
						{
							return OpCode::WriteRequest::getValue();
						}
					}
				}
			}
		}
	}
}
